src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -310,16 +336,25 @@ private async Task RefreshInitialConfiguration()
             }
         }
 
-        private async Task LoadKeyValuesRegisteredForRefresh(IDictionary<string, ConfigurationSetting> data)
+        private async Task LoadKeyValuesRegisteredForRefresh(IDictionary<string, List<ConfigurationSetting>> data)
         {
             foreach (KeyValueWatcher changeWatcher in _options.ChangeWatchers)
             {
                 string watchedKey = changeWatcher.Key;
                 string watchedLabel = changeWatcher.Label;
 
                 // Skip the loading for the key-value in case it has already been loaded
-                if (data.ContainsKey(watchedKey) && data[watchedKey].Label == watchedLabel.NormalizeNull())
+                if (data.ContainsKey(watchedKey) && data[watchedKey].Any(kv => kv.Label == watchedLabel.NormalizeNull()))
                 {
+                    int index = data[watchedKey].FindIndex(kv => kv.Label == watchedLabel.NormalizeNull());
+
+                    if (index >= 0 && index < data[watchedKey].Count() - 1)
+                    {
+                        // Move this setting to the last position because refresh registered setting should overwrite previously loaded settings

```

Comment:
>// Move this setting to the last position because refresh registered setting should overwrite previously loaded settings [](start = 24, length = 120)

If the value for a key is coming from three sources (Select, Register 1, Register 2), then the value from last source (Register 2) should always be the last one. This code snippet would instead move the value being processed to the last position.

Reply:
If a setting is loaded multiple times, the data cache always needs to be updated with the most recently loaded value at the end of the value list. 
Say we have a TestKey with labels A, B, C, and all of them are loaded by Select in order (A,B,C). 
If key with label A is loaded by Register 1 and label B is loaded by Register 2 - we want the final value to be the one corresponding to label B.
The first iteration over ChangeWatchers will change the order of settings in `data` cache to (B,C,A).
The second iteration over ChangeWatchers will change the order of settings in `data` cache to (C,A,B).

Reply:
In your example, I would expect the key to always return the value corresponding to label B, instead of returning value for A at some times and B at other times, since the latter would result in inconsistent behavior. In case you feel that we might want to reorder them in some scenario, let's discuss it over a call.